### PR TITLE
feat(profiles/shifts): tighten dietary prompt flow (#279)

### PR DESCRIPTION
**Pull request by FrankFanteev**
## Summary

Tightens the dietary-info prompt flow so humans are asked for their meal preference at the right moments, instead of only via a post-signup dashboard nudge. Three layers:

- **Soft nudge (everyone):** the dashboard "Things to do" dietary item now fires whenever `DietaryPreference` is empty — even before any signup — with no-shift copy. Once a qualifying signup exists it switches to the existing copy.
- **Hard gate (qualifying signups):** signing up for a 6+ hour / all-day ("cantina-qualifying") shift while `DietaryPreference` is empty now redirects to the dietary form first, then replays the original signup. A banner + disabled Sign-Up buttons on `/Shifts` and `/Shifts/Mine` reinforce this for anyone with a qualifying signup but no preference.
- **Meal pref next to general info:** `DietaryPreference` + `Allergies` now also appear under General Information on `/Profile/Me/Edit`. The dedicated `DietaryMedical` page still owns intolerances + medical conditions (medical = GDPR Art. 9, restricted visibility); the Edit save touches only meal-pref + allergies and never clobbers them.

This supersedes #713. It is ~90% net-new work; #279 was the originating reference.

## Scope notes

- **Cantina weekly roster (#36) is intentionally NOT in this PR.** It depends on the cross-section `IServiceRead` re-architecture that main's hard-rules now require (read names via `IUserServiceRead`, no cross-section repository reach), which is Peter-gated. It will be split into its own PR.
- Branched fresh off current `main` (merge integrated), so this diff is dietary-only.

## Test plan

- [x] `dotnet build Humans.slnx` — green
- [x] `dotnet test Humans.slnx` — all green **except** 2 pre-existing `StorePayActionTests` failures, verified failing identically on unmodified `origin/main` (unrelated Store/Stripe auth-redirect; this branch touches no Store/Stripe files)
- [x] New coverage: `ShiftsControllerDietaryGateTests`, `ProfileControllerDietaryMedicalReplayTests`, `ProfileControllerEditTests` (dietary GET/POST), `ThingsToDoViewComponentDietaryGateTests`, `DietaryMissingBannerViewComponentTests`, `ShiftSignupServicePeekRangeShiftsTests`, `ShiftTests`
- [x] Manual smoke on preview env: empty-dietary human sees soft nudge pre-signup; qualifying signup → redirect → form → replay; meal-pref round-trips on `/Profile/Me/Edit` without wiping intolerances/medical

Spec: `docs/superpowers/specs/2026-05-25-dietary-prompt-tightening-design.md`
Plan: `docs/superpowers/plans/2026-05-25-dietary-prompt-tightening.md`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/Humans.Web/ViewComponents/DietaryMissingBannerViewComponent.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+public sealed class DietaryMissingBannerViewComponent : ViewComponent
+{
+    private readonly IShiftManagementService _shiftMgmt;
+
+    public DietaryMissingBannerViewComponent(IShiftManagementService shiftMgmt)
+    {
+        _shiftMgmt = shiftMgmt;
+    }
+
+    public async Task<IViewComponentResult> InvokeAsync(Guid userId)
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Web/ViewComponents/DietaryMissingBannerViewComponent.cs:25]: Fixed in 3c93320e5. Wrapped `InvokeAsync` in try/catch, injected `ILogger<DietaryMissingBannerViewComponent>`, and log the exception before returning `Content(string.Empty)` — mirrors the ThingsToDo dietary check. (Logging in the catch rather than swallowing silently, per code-review-rules "every catch must log".) Added a test covering the service-throws → empty-content path.

File: src/Humans.Web/ViewComponents/DietaryMissingBannerViewComponent.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace Humans.Web.ViewComponents;
+
+/// <summary>
+/// Renders a red banner on /Shifts and /Shifts/Mine when the user has a
+/// qualifying cantina signup but no dietary info on file. The visibility
+/// gate lives inside the component so callers can invoke it unconditionally.
+/// Spec: docs/superpowers/specs/2026-05-25-dietary-prompt-tightening-design.md
+/// </summary>
+public sealed class DietaryMissingBannerViewComponent : ViewComponent
```
[the remark this answers is not in the document]
Comment (peterdrier) [reply src/Humans.Web/ViewComponents/DietaryMissingBannerViewComponent.cs:12]: Fixed in 3c93320e5. Registered a rendered `<vc:dietary-missing-banner>` card in `Views/WidgetGallery/Index.cshtml` next to ThingsToDo — it self-gates the same way, so it renders empty unless the viewer matches the qualifying-signup + empty-dietary condition.

**Comment by peterdrier** [comment]
## PR Review — 2026-05-25

**PR**: #778 — feat(profiles/shifts): tighten dietary prompt flow (#279)
**Branch**: `feature/dietary-prompts` (FrankFanteev) → `main`
**Scope**: 42 files (~1.2k LOC code, ~1.1k tests, ~3k docs/plans/specs); head `3c93320e5`

---

### Prior Review Status
- **Banner exception handling** (Claude bot) — **FIXED** `3c93320e5`: `InvokeAsync` now wrapped in try/catch, injects `ILogger`, logs then returns empty. Thread resolved.
- **Widget Gallery registration** (Claude bot) — **FIXED** `3c93320e5`: `<vc:dietary-missing-banner>` card added. Thread resolved.
- **`code-quality` CI (dotnet format)** — **FIXED** `3c93320e5`: switch-case brace whitespace reformatted; check now green.

---

### Spec Compliance — Issue #279 (placeholder/`blocked:spec-incomplete`): **FULLY ADDRESSED**
#279 deferred real acceptance criteria; the PR supplies them as in-repo docs (`docs/features/profiles/dietary-medical-nudge.md` US-35.1–35.7 + design spec) and implements against them. Every #279 criterion is present: trigger logic (`HasQualifyingCantinaSignupAsync`), qualifying-shift definition (`Shift.QualifiesForCantinaMeal` = all-day OR ≥6h), dietary options (Omnivore/Vegetarian/Vegan/Pescatarian), collection form, GDPR Art. 9 handling (no new columns/migration; medical kept off the general Edit form).

Three additions beyond #279 — **all justified by the PR's own design doc**: soft nudge for everyone pre-signup (US-35.5), hard gate + redirect-replay + banner + disabled buttons (US-35.6), meal-pref + allergies on Profile Edit (US-35.7).

Documented divergences (not silent drift): "modal" → full-page + server redirect (considered-and-rejected in spec Non-Goals); dietary `[Required]` not enum-validated server-side (free-text column kept to avoid migrating legacy values — spec YAGNI); range-filter + flash-mapping duplication kept per the extract-on-third-copy doctrine. Cosmetic: spec doc says tests land in `Humans.Web.Tests`, they actually live in `Humans.Application.Tests`.

---

### Code Quality

**CRITICAL** — none. Hard gate is server-side; replay paths match the direct signup paths on every parameter (`actorUserId`, `isPrivileged`, `skipConflicts`); Edit-save medical-isolation correct in code; auth inherited from class-level `[Authorize]`. Layering clean (no cross-section repo reach, no EF in service/web, no cross-section EF joins). Resx parity holds across all 6 locales.

**IMPORTANT**
1. **`HasQualifyingCantinaSignupAsync` has no service-level test.** The load-bearing "future-only" filter (`GetAbsoluteEnd > now`) and fail-closed-on-no-active-event branch are only exercised via mocks that stub the return. Add DB-backed tests: past-only qualifying signup → false; no active event → false. *(`ShiftManagementService.cs:1819`)*
2. **Edit-save-doesn't-clobber-medical invariant has no controller test.** The GDPR Art. 9 guarantee (Edit POST writes only `DietaryPreference`/`Allergies`/`AllergyOtherText`) is correct in code but untested — a future refactor to `ViewModel.ApplyTo` would silently wipe `Intolerances`/`MedicalConditions` undetected. Add a regression test seeding non-empty medical/intolerances, POST Edit, assert unchanged. *(`ProfileController.cs:486`)*
3. **`GetShiftProfileAsync(includeMedical:false)` returns a mutated entity** (`MedicalConditions = null`). `AsNoTracking` keeps the DB safe, but passing that object to `UpdateShiftProfileAsync` would drop medical data — latent footgun. Add a warning comment on the method, or return a read-only projection. *(`ShiftManagementService.cs:1807`)*

**MINOR**
4. Section invariant docs not updated (`docs/sections/Shifts.md` lacks the three new service/repo methods; `Profiles.md` lacks the US-35.7 Edit-isolation invariant). The feature doc was updated; the terse section docs weren't (`post-fix-doc-check`).
5. `DietaryMedical.cshtml` / `Edit.cshtml` radios use redundant `checked="@(isSelected ? "checked" : null)"` alongside `asp-for` (tag helper already handles it). Safe (null-pattern, not the `disabled="false"` trap) but redundant.

---

### CI
- `code-quality` — **green** (after format fix).
- `build` — one **flaky** failure: `RemoveLineAsync_RemovesLine_AndRecomputesTotal` (a Budget test, unrelated to this PR) hit `Test execution timed out after 5000ms`; 3090/3092 passed. Re-running.

---

### Bottom Line
Spec-complete and architecturally clean; no blocking issues. The substantive follow-ups are two **test gaps** (#1 future-only filter, #2 the GDPR Art. 9 Edit-isolation invariant — the most valuable to close) and one **footgun comment** (#3). Docs (#4) and the redundant radio attr (#5) are minor. The only red CI is a flaky unrelated timeout.

---
*Review by Claude Code `/pr-review`*
